### PR TITLE
[Refactor] Seperate starcache from staros to avoid build staros modules if unnecessary.

### PR DESCRIPTION
**Pull request by GavinMar**
We seperate starcache from staros for the following reasons:
* At present, use staros is not enabled by default on the main branch, so the version compiled by users according to the default method cannot use DLA datacache.
* We still keep the seperate starcache library from staros, because the starrocks branch-3.0 depend on starcache, while the target staros doesn't depened and include it.
* It seems a bit heavy overall to make the starcache  depend on staros. In order to use cache, users must build it with staros. This in turn leads to the introduction of some other modules and dependencies of staros.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by DorianZheng** [comment]
does staros still contain starcache? if yes, will their symbols conflict?

**Review by DorianZheng — APPROVED** [review]
left a comment, rest LGTM

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: build.sh [hunk 1]
```diff
@@ -177,6 +177,12 @@ if [[ "${WITH_CACHELIB}" == "ON" && ! -f ${STARROCKS_THIRDPARTY}/installed/cache
     exit 1
 fi
 
+# Temporarily turn off the starcache by default to avoid frequent thirdparty update until the current
+# starcache version is stable.
+if [[ -z ${WITH_STARCACHE} ]]; then
+  WITH_STARCACHE=OFF
```
Comment (kevincai) [inline build.sh:183]: indention

File: be/CMakeLists.txt [hunk 2]
```diff
@@ -402,6 +402,11 @@ if (${WITH_CACHELIB} STREQUAL "ON")
     endforeach()
 endif()
 
+if (${WITH_STARCACHE} STREQUAL "ON")
```
Comment (kevincai) [inline be/CMakeLists.txt:405]: what if WITH_STARCACHE=OFF and USE_STAROS=ON, will the build break?

File: run-be-ut.sh [hunk 3]
```diff
@@ -149,6 +155,8 @@ if [ "${USE_STAROS}" == "ON"  ]; then
     STARLET_INSTALL_DIR=${STARROCKS_THIRDPARTY}/installed/starlet
   fi
   export STARLET_INSTALL_DIR
+  # To keep the behavior same with before util the the starcache is stable and we will turn on it by default.
```
Comment (kevincai) [inline run-be-ut.sh:158]:
```suggestion
  # To keep the behavior same as before util the the starcache is stable and we will turn on it by default.
```

File: thirdparty/vars-ubuntu22-aarch64.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#####################################################
+# Download url, filename and unpaced filename
+# of all thirdparties
+# 
+# vars-${arch}.sh defines the thirdparties that are
+# architecure-related.
+#####################################################
+
+# starcache
+STARCACHE_DOWNLOAD="https://cdn-thirdparty.starrocks.com/starcache/v2.0/ubuntu-aarch64/starcache.tar.gz"
```
Comment (kevincai) [inline thirdparty/vars-ubuntu22-aarch64.sh:25]: wrong link?

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by GavinMar** [comment]
> does staros still contain starcache? if yes, will their symbols conflict?



> does staros still contain starcache? if yes, will their symbols conflict?

With same starcache version, no symbol conflict.

File: be/CMakeLists.txt (hunk 2 above)
Comment (GavinMar) [reply be/CMakeLists.txt:405]: > what if WITH_STARCACHE=OFF and USE_STAROS=ON, will the build break?

Update

File: thirdparty/vars-ubuntu22-aarch64.sh (hunk 4 above)
Comment (GavinMar) [reply thirdparty/vars-ubuntu22-aarch64.sh]: > wrong link?

Fixed

**Review by dirtysalt — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by dirtysalt — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**